core/templates/pages/story-editor-page/editor-tab/story-node-editor.component.ts

```diff
@@ -203,13 +221,51 @@ export class StoryNodeEditorComponent implements OnInit, OnDestroy {
       this.storyUpdateService.setStoryNodeDescription(
         this.story, this.nodeId, newDescription);
       this.currentDescription = newDescription;
+      this.updateCurrentNodeIsPublishable();
     }
   }
 
+  updatePlannedPublicationDate(dateString: string | null): void {
+    let newPlannedPublicationDate = dateString && dateString.length ? new Date(
+      dateString) : null;
+
+    if (newPlannedPublicationDate !== this.plannedPublicationDate) {
+      if (newPlannedPublicationDate) {
+        let currentDateTime = new Date();
+        if (newPlannedPublicationDate.getTime() < currentDateTime.getTime()) {
+          this.plannedPublicationDateIsInPast = true;
+          if (this.plannedPublicationDate) {
+            this.storyUpdateService.setStoryNodePlannedPublicationDateMsecs(
+              this.story, this.nodeId, null);
+          }
+          this.plannedPublicationDate = null;
+          this.editablePlannedPublicationDate = null;
+          setTimeout(() => {
+            this.plannedPublicationDateIsInPast = false;
+          }, 5000);
```

Comment:
What does this number represent? Should we have a named variable?

Reply:
This is just a time interval of 5 sec for a toast showing an error message.

Reply:
Sounds good, please create a named constant in that case so it is clear to future devs about what it represents.